### PR TITLE
Remove half-baked Storage Driver IPC support

### PR DESCRIPTION
**Pull request by stevvooe**
This removes documentation and coding related to IPC based storage driver
plugins. The existence of this functionality was an original feature goal but
is now not-maintained and actively confusing incoming contributions. We will
likely explore some driver plugin mechanism in the future but we don't need
this laying around in the meantime.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

**Comment by stevvooe** [comment]
CC @BrianBland @dmcgowan @AndreyKostov @dmp42

File: registry/storage/driver/s3/s3_test.go [hunk 1]
```diff
@@ -74,79 +75,64 @@ func init() {
...
-type S3DriverSuite struct {
-	Constructor S3DriverConstructor
-	testsuites.SkipCheck
+	}, skipS3)
 }
 
-func (suite *S3DriverSuite) SetUpSuite(c *check.C) {
-	if reason := suite.SkipCheck(); reason != "" {
-		c.Skip(reason)
+func TestEmptyRootList(t *testing.T) {
```
Comment (lebauce) [inline registry/storage/driver/s3/s3_test.go]: It seems this test is not run. With this code :
```
OK: 0 passed, 33 skipped
PASS
ok  	github.com/docker/distribution/registry/storage/driver/s3	0.003s
```

with master :
```
OK: 0 passed, 34 skipped
PASS
ok  	github.com/docker/distribution/registry/storage/driver/s3	0.003s
```

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
Need another peak here: @RichardScothern

**Comment by BrianBland** [comment]
LGTM

**Pull request merged**